### ISSUE TITLE
OSX: 1.5.0-beta1 `MULTILIB=ON` is broken

### ISSUE THREAD

**Issue by JohanEngelen** [ldc-developers/ldc#2371]
(at least on macOS) `cmake -DMULTILIB=ON` is broken since cf03f5c040282643ea2ad17483f2f02b133061d1 .

This prevents building the macOS binaries of 1.5.0-beta1.

The errors are:
```
CMake Error at runtime/CMakeLists.txt:528 (target_link_libraries):
  Target "druntime-ldc-shared" of type UTILITY may not be linked into another
  target.  One may link only to STATIC or SHARED libraries, or to executables
  with the ENABLE_EXPORTS property set.
Call Stack (most recent call first):
  runtime/CMakeLists.txt:600 (build_runtime_libs)
  runtime/CMakeLists.txt:840 (build_runtime_variant)
  runtime/CMakeLists.txt:865 (build_test_runner_variant)
  runtime/CMakeLists.txt:870 (build_test_runner_variants)


CMake Error at runtime/CMakeLists.txt:814 (target_link_libraries):
  Target "druntime-ldc" of type UTILITY may not be linked into another
  target.  One may link only to STATIC or SHARED libraries, or to executables
  with the ENABLE_EXPORTS property set.
Call Stack (most recent call first):
  runtime/CMakeLists.txt:855 (build_test_runners)
  runtime/CMakeLists.txt:865 (build_test_runner_variant)
  runtime/CMakeLists.txt:870 (build_test_runner_variants)


CMake Error at runtime/CMakeLists.txt:814 (target_link_libraries):
  Target "druntime-ldc-shared" of type UTILITY may not be linked into another
  target.  One may link only to STATIC or SHARED libraries, or to executables
  with the ENABLE_EXPORTS property set.
Call Stack (most recent call first):
  runtime/CMakeLists.txt:859 (build_test_runners)
  runtime/CMakeLists.txt:865 (build_test_runner_variant)
  runtime/CMakeLists.txt:870 (build_test_runner_variants)


CMake Error at runtime/CMakeLists.txt:528 (target_link_libraries):
  Target "druntime-ldc-debug-shared" of type UTILITY may not be linked into
  another target.  One may link only to STATIC or SHARED libraries, or to
  executables with the ENABLE_EXPORTS property set.
Call Stack (most recent call first):
  runtime/CMakeLists.txt:600 (build_runtime_libs)
  runtime/CMakeLists.txt:840 (build_runtime_variant)
  runtime/CMakeLists.txt:866 (build_test_runner_variant)
  runtime/CMakeLists.txt:870 (build_test_runner_variants)
```

**Comment by JohanEngelen** [comment]
( @kinke @joakim-noah )

**Comment by JohanEngelen** [comment]
(using cmake 3.5, now installing latest cmake to see whether that is causing troubles)

**Comment by kinke** [comment]
I totally fail to see where https://github.com/ldc-developers/ldc/blob/master/runtime/CMakeLists.txt#L488 could possibly go wrong and druntime end up as 'utility' instead of a static/shared library.

**Comment by kinke** [comment]
Right, forgot the OSX-specific libs merging step with renamed normal lib targets (https://github.com/ldc-developers/ldc/blob/master/runtime/CMakeLists.txt#L691). Dammit.

**Comment by kinke** [comment]
=> #2372

**Comment by JohanEngelen** [comment]
Fixed by #2372
